esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2218,32 +2218,39 @@ void CFileSprayEx::getDropZoneInfoByIP(const char* ip, const char* destFileIn, S
     if (!env)
         return;
 
-    Owned<IConstMachineInfo> machine = env->getMachineByAddress(ip);
-    if (!machine)
-    {
-        IpAddress ipAddr;
-        ipAddr.ipset(ip);
-        if (!ipAddr.isLocal())
-            return;
-        machine.setown(env->getMachineForLocalHost());
-        if (!machine)
-            return;
-    }
-    SCMStringBuffer computer, directory, maskBuf;
-    machine->getName(computer);
-    if (!computer.length())
-        return;
-
-    Owned<IConstDropZoneInfo> dropZone = env->getDropZoneByComputer(computer.str());
-    if (!dropZone)
-        return;
-    dropZone->getDirectory(directory);
+    SCMStringBuffer directory;
+    Owned<IConstDropZoneInfo> dropZone;
 
     StringBuffer destFile;
     if (isAbsolutePath(destFileIn))
+    {
         destFile.set(destFileIn);
+        dropZone.setown(env->getDropZoneByAddressPath(ip, destFile.str()));
+        if (!dropZone)
+            return;
+        dropZone->getDirectory(directory);
+    }
     else
     {
+        SCMStringBuffer computer;
+        Owned<IConstMachineInfo> machine = env->getMachineByAddress(ip);
+        if (!machine)
+        {
+            IpAddress ipAddr;
+            ipAddr.ipset(ip);
+            if (!ipAddr.isLocal())
+                return;
+            machine.setown(env->getMachineForLocalHost());
+            if (!machine)
+                return;
+        }
+        machine->getName(computer);
+        if (!computer.length())
+            return;
+        dropZone.setown(env->getDropZoneByComputer(computer.str()));
```

Comment:
This is a legacy method. While we can have more than one DZ on same IP it returns with the default/first (in environment.xml) DZ.

If the aim of this function call is to determine whether there is a DZ on given IP, then it is fine.

If the aim to select a particular DZ then there are two methods:
1. use getDropZoneByComputer(const char * computer, const char * dzname)
2. use Owned<IConstDropZoneInfoIterator> zoneIt= getDropZoneIteratorByComputer(const char * computer)
   and iterate through the DZs with 
   ForEach(*zoneIt) { ... }